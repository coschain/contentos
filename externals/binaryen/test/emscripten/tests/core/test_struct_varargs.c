#include <stdio.h>
#include <stdarg.h>

struct A {
  int x;
};

struct B {
  double x;
};

void foo(int unused, ...)
{
  va_list vl;
  va_start(vl, unused);
  struct A a = va_arg(vl, struct A);
  struct B b = va_arg(vl, struct B);
  va_end(vl);

  printf("%d\n", a.x);
  printf("%f\n", b.x);
}

int main() {
  struct A a = {
    .x = 42,
  };
  struct B b = {
    .x = 42.314,
  };
  foo(0, a, b);
  return 0;
}

