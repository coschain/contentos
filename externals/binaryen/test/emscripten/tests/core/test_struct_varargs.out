42
42.314000
